assets/src/hooks/useAlerts.ts

```diff
@@ -0,0 +1,89 @@
+import { Channel, Socket } from "phoenix"
+import { useEffect, useState } from "react"
+import { array, assert, object, string } from "superstruct"
+import { equalByElements } from "../helpers/array"
+import { reload } from "../models/browser"
+import { ByRouteId, RouteId } from "../schedule"
+
+const subscribe = (
+  socket: Socket,
+  routeId: RouteId,
+  updateAlerts: React.Dispatch<React.SetStateAction<ByRouteId<string[]>>>
+): Channel => {
+  const handleAlerts = (alertsData: unknown): void => {
+    assert(
+      alertsData,
+      object({
+        data: array(string()),
+      })
+    )
+
+    updateAlerts((oldAlerts) => {
+      const newAlerts: ByRouteId<string[]> = {
+        [routeId]: alertsData.data,
+        ...oldAlerts,
+      }
+
+      return newAlerts
+    })
+  }
+
+  const topic = `alerts:route:${routeId}`
+  const channel = socket.channel(topic)
+
+  channel.on("alerts", handleAlerts)
+
+  channel.on("auth_expired", reload)
+
+  channel
+    .join()
+    .receive("ok", handleAlerts)
+    // eslint-disable-next-line no-console
+    .receive("error", ({ reason }) => console.error("join failed", reason))
+    .receive("timeout", reload)
+
+  return channel
+}
+
+const useAlerts = (
+  socket: Socket | undefined,
+  routeIds: RouteId[]
+): ByRouteId<string[]> => {
+  const [alertsByRoute, setAlertsByRoute] = useState<ByRouteId<string[]>>({})
+  // eslint-disable-next-line react/hook-use-state
+  const [, setChannelsByRouteId] = useState<ByRouteId<Channel>>({})
```

Comment:
Since `channelsByRouteId` is only ever referenced using the update function form of `setChannelsByRouteId`, we don't actually need to pull that variable out of the tuple. As you can see, though, this triggered a hooks ESLint rule. I'm not sure if there's a better way to handle this sort of case where we don't need to reference the state directly (but we do need it so that we know which channels need to be added / removed), but I'm open to suggestions.